src/java.base/unix/native/libnio/fs/UnixNativeDispatcher.c

```diff
@@ -71,6 +72,98 @@
 #define readdir64 readdir
 #endif
 
+#if defined(__linux__)
+// Account for the case where we compile on a system without statx
+// support. We still want to ensure we can call statx at runtime
+// if the runtime glibc version supports it (>= 2.28). We do this
+// by defining binary compatible statx structs in this file and
+// not relying on included headers.
+
+#ifndef __GLIBC__
+// Alpine doesn't know these types, define them
+typedef unsigned int       __uint32_t;
+typedef unsigned short     __uint16_t;
+typedef unsigned long int  __uint64_t;
+#endif
```

Comment:
Not sure what changed on the lastest merge from master, but I was getting compile failures on Apine without this. It's a no-op for glibc. @AlanBateman @bplb Still ok?

Reply:
> Not sure what changed on the lastest merge from master, but I was getting compile failures on Apine without this. It's a no-op for glibc. @AlanBateman @bplb Still ok?

Yes, I don't have any other comments and I see you've dropped the enablePreview.

Reply:
Likewise. Thanks.